### ISSUE TITLE
bcftools merge invalid pointer and core dump

### ISSUE THREAD

**Issue by pishotta** [samtools/bcftools#193]
<pre>When issuing the command (and similar):
/bcftools/bcftools merge --use-header h.txt -O v -o y.vcf UBC02*.vcf.gz
which processes about 5 files i get:

> *** glibc detected *** /root/bcftools/bcftools: free(): invalid pointer: 0x00000000019d24e0 > > > > > > ***======= Backtrace: =========
> /lib64/libc.so.6(+0x75e76)[0x7f67430b5e76]
> /root/bcftools/bcftools[0x411f01]
> /root/bcftools/bcftools[0x416fdd]
> /root/bcftools/bcftools[0x417963]
> /root/bcftools/bcftools[0x417f58]
> /root/bcftools/bcftools[0x4185e6]
> /lib64/libc.so.6(__libc_start_main+0xfd)[0x7f674305ed5d]
> /root/bcftools/bcftools[0x403419]
> ======= Memory map: ========
> 00400000-004f6000 r-xp 00000000 fd:00 263735                             /root/bcftools/bcftools
> 006f6000-006f9000 rw-p 000f6000 fd:00 263735                             /root/bcftools/bcftools
> 009a0000-019f1000 rw-p 00000000 00:00 0                                  [heap]
> 3f8b400000-3f8b415000 r-xp 00000000 fd:00 392679                         /lib64/libz.so.1.2.3
> 3f8b415000-3f8b614000 ---p 00015000 fd:00 392679                         /lib64/libz.so.1.2.3
> 3f8b614000-3f8b615000 r--p 00014000 fd:00 392679                         /lib64/libz.so.1.2.3
> 3f8b615000-3f8b616000 rw-p 00015000 fd:00 392679                         /lib64/libz.so.1.2.3
> 3f96000000-3f96016000 r-xp 00000000 fd:00 392586                         /lib64/libgcc_s-4.4.7-20120601.so.1
> 3f96016000-3f96215000 ---p 00016000 fd:00 392586                         /lib64/libgcc_s-4.4.7-20120601.so.1
> 3f96215000-3f96216000 rw-p 00015000 fd:00 392586                         /lib64/libgcc_s-4.4.7-20120601.so.1
> 7f6742918000-7f6742a19000 rw-p 00000000 00:00 0
> 7f6742a19000-7f6742b1a000 rw-p 00000000 00:00 0
> 7f6742b1a000-7f6742c1b000 rw-p 00000000 00:00 0
> ...etc...

Although it works fine when fewer files (2 or 3) are processed.  Memory usage problem?
Too much?  To me, the dump is uninformative.  Have also tried with current development version
Version: 1.1-141-g10cb60e (using htslib 1.1-125-g3768707)

Thanks.  -- Fred P.
</pre>

**Comment by pd3** [comment]
Hi, is there any chance you could provide an example so I can investigate? The core dump does not give enough information.

**Comment by pishotta** [comment]
Not easily.   The input files are big.  Let me narrow down the problem a bit to see if I can get a reproducible error on fewer or smaller input files.

**Comment by pishotta** [comment]
<pre>Not resolved, but I think I found the triggering issue for the coredumps,
I was originally having trouble when the input vcfs had headers varying on the INFO lines
>  ##INFO=&lt;ID=FR,Number=.,Type=String,Description="Reason why the variant was filtered."&gt;, 
    and
>  ##INFO=&lt;ID=FR,Number=1,Type=String,Description="Reason why the variant was filtered."&gt;
Specifically, the value of "Number", most records in the files having just "." as a value, but a few in
some files having ".,SOMESTRING" as a value.  Attempts to merge, issued an error about header
definition being of different lengths. First workaround attempt was to --use-header to get consistent
headers.  For whatever reason, some combinations of those values in those input files triggered
coredumps.  FINALLY, after much putzing, I modified the header lines in the 80 input files so they
all read:
> ##INFO=&lt;ID=FR,Number=.,Type=String,Description="Reason why the variant was filtered."&gt;
and the merge completed properly ("." meaning "unlimited" in this context). Problem doesn't seem
related to memory usage or N of files, but there's still a bug or failed error trap in there somewhere.
Found a work-around, but beware.

-- Fred P.
</pre>
